File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        elif x < 10:
            return True
        revX = 0
        tempX = x
        while(tempX):
            revX = (tempX%10) + (revX * 10)
            tempX //= 10
        if revX == x:
            return True
        return False

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int temp = x;
        int reversedInteger = 0;

        while (temp > 0) {
            reversedInteger *= 10;
            reversedInteger += temp%10;
            temp /= 10;
        }

        return reversedInteger == x;
    }
}